lib/api/CInferenceModelDefinition.cc

```diff
@@ -380,6 +392,48 @@ std::string CInferenceModelDefinition::jsonString() {
     return resultString;
 }
 
+std::stringstream CInferenceModelDefinition::jsonStringCompressedFormat() const {
+    std::stringstream compressedStream;
+    using TFilteredOutput = boost::iostreams::filtering_stream<boost::iostreams::output>;
+    std::string modelDefinitionStr{jsonString()};
+    {
+        TFilteredOutput outFilter;
+        outFilter.push(boost::iostreams::gzip_compressor());
+        outFilter.push(core::CBase64Encoder());
+        outFilter.push(compressedStream);
+        outFilter << modelDefinitionStr;
+        outFilter.flush();
+    }
+    return compressedStream;
+}
+
+void CInferenceModelDefinition::addToDocumentCompressed(TRapidJsonWriter& writer) const {
+    std::stringstream compressedString{jsonStringCompressedFormat()};
+    rapidjson::Value definitionChunksArray{writer.makeArray()};
+    std::streamsize processed{0};
+    compressedString.seekg(0, compressedString.end);
+    std::streamsize remained{compressedString.tellg()};
+    compressedString.seekg(0, compressedString.beg);
+    std::size_t utf16Length{static_cast<std::size_t>(remained)}; // since we use base64 encoding
```

Comment:
The way this is done at the moment is not compatible with streaming into the compressor.  But it seems like the output format is a document containing an array of objects, each of which knows the overall length.  If the format could be changed to just have the overall length at the very end in a separate field outside the array then that would work with streaming.  The streamer class would keep track of the length so far, and then at the very end you'd get this and add a length field.

Reply:
We discussed this offline. Needing the whole compressed state upfront was for a consistency check in the Java, but  we can do this in a different way, so we do think it is is possible to stream the chunked documents out. This will require a small change to the schema. It is still TBD whether we'll make these changes as a follow up.

Reply:
We changed the format to pass sequential  `doc_num`s and `eos` flag for the last chunk.